### DISCUSSION TITLE
[Feature] Sync certain time range or all pictures to mobile

### DISCUSSION THREAD

**Discussion by jakob42** [immich-app/immich#5695] [Feature Request]
### The feature

My wife and I moved from a smb share and scripts to immich and are rather happy with it. (Thanks to all involved!) I know there is partner sharing now, but we are using a single account together. Since we dont always have stellar mobile connection, we would like to have the pictures offline available.

I would love to see a feature in the mobile app that lets me sync all the pictures or optionally the last x month to my phone while on wifi. (And deletes them locally when they leave the time range)

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by zhaolin-st** [comment]
Hi jakob,

Thank you for raising this feature request. It resonates with my own needs for Immich.

I would like to add my support for a similar feature but with a slight variation. My requirement is to have a synchronization feature in Immich that allows the mobile client to keep a complete copy of all photos from the server, not just accessible through the app, but stored locally on the device. This would ensure that I can access and use my photos anytime, even without internet access.

You can find my initial request here: https://github.com/immich-app/immich/discussions/11344#discussioncomment-10147437. Implementing this feature would greatly enhance the usability of Immich for users who often find themselves without reliable mobile connections.

Thank you for considering this enhancement.

**Comment by Dracco61** [comment]
As stated by @zhaolin-st I would love a feature that allows the Immich mobile client to store a local copy of server photos. While this can be set up via workarounds, it would greatly improve usability and accessibility for users without as much technical know-how on VPNs or apps like Syncthing.

**Comment by AppyxDaniel** [comment]
Same here, in my case I migrated my stuff to Immich using my PC and then I wanted to have my files available offline on my phone.
Currently one has to select all assets for download manually, which is quite a pain. 

Also, if one gets a new phone one might want to download the assets from the last year or some other range to have the assets available offline.
